### PR TITLE
EWS v2 - enhancement

### PR DESCRIPTION
**Pull request by IkaDemisto**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/19373
fixes: https://github.com/demisto/etc/issues/19422
fixes: https://github.com/demisto/etc/issues/19423

## Description
Added new command `ews-get-items-as-eml` and improved `ews-search-mailbox` command.

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [x] Documentation (with link to it)
- [x] Code Review

## Dependencies
Mention the dependencies of the entity you changed as given from the precommit hooks in checkboxes, and tick after tested them.
- [x] Search And Delete Emails - Generic
- [x] Get Original Email - Generic
- [x] Phishing Investigation - Generic
- [x] Process Email - Generic
- [x] Office 365 Search and Delete
- [x] Search And Delete Emails - EWS
- [x] Process Email - EWS
- [x] Phishing Investigation - Generic v2
- [x] Get Original Email - EWS

## Technical writer review
Mention and link to the files that require a technical writer review.
- [x] [YAML file](https://github.com/demisto/content/blob/5728c1a82b9b3c56040772ce3cf85abf0ece1c3c/Integrations/EWSv2/EWSv2.yml)
- [x] [CHANGELOG](https://github.com/demisto/content/blob/5728c1a82b9b3c56040772ce3cf85abf0ece1c3c/Integrations/EWSv2/CHANGELOG.md)

### CODE REVIEW

File: Integrations/EWSv2/EWSv2.py [hunk 1]
```diff
@@ -1851,6 +1877,30 @@ def mark_item_as_read(item_ids, operation='read', target_mailbox=None):
                                 marked_items)
 
 
+def get_item_as_eml(item_id, target_mailbox=None):
+    account = get_account(target_mailbox or ACCOUNT_EMAIL)
+    item = get_item_from_mailbox(account, item_id)
+
+    if not isinstance(item, Message):
+        return_error(item)
```
Comment (yaakovi) [inline Integrations/EWSv2/EWSv2.py]: what is the type of item here?
return_error should have a readable message

File: Integrations/EWSv2/EWSv2.py [hunk 2]
```diff
@@ -2017,6 +2069,8 @@ def main():
 
         if demisto.command() == 'fetch-incidents':
             raise
+        if demisto.command() == 'ews-search-mailbox' and isinstance(e, ValueError):
+            return_error("Selected invalid field, please specify valid field name.")
```
Comment (yaakovi) [inline Integrations/EWSv2/EWSv2.py]: nice, you should specify what is the invalid field (you have it in `e`)

File: Integrations/EWSv2/EWSv2.yml [hunk 3]
```diff
@@ -417,6 +417,15 @@ script:
       name: message-id
       required: false
       secret: false
+    - default: false
+      defaultValue: all
+      description: A CSV list if fields to retrieve.
```
Comment (yaakovi) [inline Integrations/EWSv2/EWSv2.yml]:
```suggestion
      description: A CSV list of fields to retrieve.
```

File: Integrations/EWSv2/EWSv2.py (hunk 2 above)
Comment (IkaDemisto) [reply Integrations/EWSv2/EWSv2.py]: The error will look like: `Selected invalid field, please specify valid field name. Unknown fieldname 'subjec' on folders '[Folder(Root(<exchangelib.account.Account object at 0x7f9413752a90>, u'[self]', u'root', 16, 0, 88, None, 'AAMkADQ0NmFkODFkLWQ4MDEtNDE4Mi1hN2NkLThmZjdmNTZjNTMxNwAuAAAAAAA4kxh+ed3JTJPMPXU3wX3aAQCyyVyFtlsUQZfBJebinpkUAAAAAAEBAAA=', 'AQAAABYAAACyyVyFtlsUQZfBJebinpkUAAAfuQHa'), 'Files', 0, 0, 0, 'IPF.Files', 'AAMkADQ0NmFkODFkLWQ4MDEtNDE4Mi1hN2NkLThmZjdmNTZjNTMxNwAuAAAAAAA4kxh+ed3JTJPMPXU3wX3aAQCyyVyFtlsUQZfBJebinpkUAAAAAAFVAAA=', 'AQAAABYAAACyyVyFtlsUQZfBJebinpkUAABHP3Dt')]' in only()` with concatenating e message

File: Integrations/EWSv2/EWSv2.py [hunk 4]
```diff
@@ -1424,32 +1426,54 @@ def search_items_in_mailbox(query=None, message_id=None, folder_path='', limit=1
...
+    else:
+        searched_items_result = map(
+            lambda item: parse_item_as_dict(item, account.primary_smtp_address, camel_case=True, compact_fields=False),
+            items)
+        searched_items_result = [
+            {k: v for (k, v) in i.iteritems()
+             if k in keys_to_camel_case(restricted_fields)} for i in searched_items_result]
+
+        for item in searched_items_result:
+            item['itemId'] = item.pop('id', '')
```
[the remark this answers is not in the document]
Comment (IkaDemisto) [reply Integrations/EWSv2/EWSv2.py]: Yes, `parse_item_as_dict` returns additional data (all the fields where `getattr` is called) and the idea is give the user the control over the fields are set to context. From the api `id` is returned and switched to `itemId` in order to append correctly the result to the context. `CONTEXT_UPDATE_EWS_ITEM = "EWS.Items(val.{0} == obj.{0} || (val.{1} && obj.{1} && val.{1} == obj.{1}))".format(ITEM_ID,
                                                                                                               MESSAGE_ID)`, where ITEM_ID is itemId...

File: Integrations/EWSv2/EWSv2.py (hunk 1 above)
Comment (IkaDemisto) [reply Integrations/EWSv2/EWSv2.py]: Not necessary, removed it.

**Comment by yaakovi** [comment]
@kirbles19 please review change log and new additional command in yml

**Comment by kirbles19** [comment]
@IkaDemisto Done.

**Pull request merged**